tests/span_tests.cpp

```diff
@@ -577,73 +447,40 @@ TEST(span_test, from_container_constructor)
     const std::string cstr = "hello";
 
     {
-#ifdef CONFIRM_COMPILATION_ERRORS
+#ifdef FAIL_ON_SOME_PLATFORMS
```

Comment:
I introduced this (not set macro) check because these checks sometimes succeed and sometimes fail.